### PR TITLE
Updates to ad consents page

### PR DESCRIPTION
**Pull request by walaura**
## What does this change?
Adds all the missing UI bits and bobs to the advert consents page (#19649), namely:

– A submit button, this has rejiggered most of the event handling since the last time
– A FeedbackFlashBox element, it appears after clicking save to give feedback the changes were saved. It also uses `ref`. Sorry! But this page is a bit pressed for time.
– Some CSS that looks closer to the final version.

## What is the value of this and can you measure success?
Preemptively split up what would be a massive PR into smaller ones. This is not the final page so quality is a bit all over the board.

## Screenies
screen shot 2018-05-18 at 4 48 01 pm

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/ad-prefs.js [hunk 1]
```diff
@@ -80,31 +73,100 @@ class ConsentBox extends Component<
...
+            changesPending: false,
+            consentsWithState: allAdConsents.map((consent: AdConsent) => ({
+                consent,
+                state: getAdConsentState(consent),
+            })),
+        };
+    }
+
+    onUpdate(consentId, state: AdConsentState) {
+        const consentsWithState = { ...this.state }.consentsWithState;
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/identity/ad-prefs.js]: Is destructuring necessary here? Could we not just write `this.state. consentsWithState`.

Comment (sndrs) [reply static/src/javascripts/projects/common/modules/identity/ad-prefs.js]:
```js
const { consentsWithState } = this.state;
```
?

File: static/src/javascripts/projects/common/modules/identity/ad-prefs/FeedbackFlashBox.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// @flow
+import React, { Component } from 'react';
+
+type FeedbackFlashBoxProps = {
+    children?: any,
+};
+
+class FeedbackFlashBox extends Component<
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/identity/ad-prefs/FeedbackFlashBox.js]: Could we add return types to this component's functions?

File: static/src/javascripts/projects/common/modules/identity/ad-prefs.js (hunk 1 above)
Comment (walaura) [reply static/src/javascripts/projects/common/modules/identity/ad-prefs.js]: I don't want to mutate the original state, just to grab a copy of it. I know it seems a bit pointless considering I'm using the copy to define the new state but I believe this is best practice

**Comment by walaura** [comment]
[Ads consent tool – save button](https://trello.com/c/obwnlOTh/647-ads-consent-tool-save-button)

File: static/src/stylesheets/module/identity/_ad-prefs.scss [hunk 3]
```diff
@@ -0,0 +1,39 @@
+.identity-ad-prefs-manager {
+
+    fieldset {
+        border: 0;
+        margin: 0;
+        padding: 0;
+        legend {
+            position: absolute;
+            left: -2000px;
```
Comment (GHaberis) [inline static/src/stylesheets/module/identity/_ad-prefs.scss]: Are you positioning the legend offscreen to hide it? If so can you use the `u-h` class on this element instead? This'll mean screen readers will pick it up but it won't be visible.

**Review by GHaberis — APPROVED** [review]
looks good 👍

**Comment by Jester232** [comment]
LAndon

**Pull request merged**